source/secured/symmetric.d

```diff
@@ -36,45 +36,32 @@ public enum SymmetricAlgorithm : ubyte {
 }
 
 public immutable struct EncryptedData {
+    public immutable ubyte[] iv;
     public immutable ubyte[] cipherText;
-    public immutable SymmetricMetadata metadata;
+    public immutable ubyte[] authTag;
 
     @safe public this(const string encoded) {
```

Comment:
API Improvement:
1. Add `SymmetricAlgorithm` and `HashAlgorithm` with defaults.
2. Store those in the type as was done in SymmetricMetadata.
3. Then pass those values through to `encrypt_ex/decrypt_ex`.